### PR TITLE
Persist deployment and workspace IDs for flow command

### PR DESCRIPTION
**Pull request by feluelle**
## Description

This PR fixes the following issues:

* whitespace issues in flow dockerfile and deploy cmd
* reset of deployment config values
* e2e tests to run on latest version

## 📋 Checklist

- [x] Rebased from the main (or release if patching) branch (before testing)
- [x] Ran `make test` before taking out of draft
- [x] Ran `make lint` before taking out of draft
- [ ] Added/updated applicable tests
- [ ] Tested against [Astro-API](https://github.com/astronomer/astro/) (if necessary).
- [ ] Tested against [Houston-API](https://github.com/astronomer/houston-api/) and [Astronomer](https://github.com/astronomer/astronomer/) (if necessary).
- [ ] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

**Review by tatiana — APPROVED** [review]

File: cmd/sql/e2e/flow_test.go [hunk 1]
```diff
@@ -33,8 +33,9 @@ func chdir(t *testing.T, dir string) func() {
 }
 
 func execFlowCmd(args ...string) error {
+	// Set the astro-cli version to a version compatible with the latest sql-cli version for e2e tests
+	version.CurrVersion = "1.10.0"
```
Comment (neel-astro) [inline cmd/sql/e2e/flow_test.go:37]: nit: is there a way to not need to update this on all releases? isn't it always equal to the latest CLI release version as per semver?

**Review by neel-astro — APPROVED** [review]

File: cmd/sql/e2e/flow_test.go (hunk 1 above)
Comment (feluelle) [reply cmd/sql/e2e/flow_test.go:37]: Yes, there is a way, but currently, our config is allowing specific versions to be used for sql-cli versions. See [here](https://github.com/astronomer/astro-sdk/blob/astro-cli/sql-cli/config/astro-cli.json).

What we probably want is simply to have an entry for `latest` where we don't set any upper bound. Currently, the `1.10.0` entry is such an entry, but we might have to upper-bound it in the future.

We have the same issue if we are on a branch. The airflow version is a `SNAPSHOT-*`. We probably want to skip the compatibility matrix in these cases, maybe also for e2e tests? WDYT?

Comment (neel-astro) [reply cmd/sql/e2e/flow_test.go:37]: hmmm, anything with `SNAPSHOT-*` can be treated as the latest CLI version from SDK POV

Comment (feluelle) [reply cmd/sql/e2e/flow_test.go:37]: Sorry, I meant the astro-cli version - not airflow version.

**Pull request merged**